src/App.tsx

```diff
@@ -1,24 +1,281 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import * as postService from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 10876;
+const emptyTodo = {
+  id: 0,
+  title: '',
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<string>('all');
+  const [isHideError, setIsHideError] = useState<boolean>(true);
+  const [error, setError] = useState<string>('');
+  const [query, setQuery] = useState<string>('');
+  const [tempTodo, setTempTodo]
+    = useState<{ id: number, title: string }>(emptyTodo);
+  const [isTodoLoaded, setIsTodoLoaded] = useState<boolean>(false);
+  const [deleteTodoIds, setDeleteTodoIds] = useState([0]);
+
+  const handleErrorMessage = (message: string) => {
+    setIsHideError(false);
+    setError(message);
+
+    setTimeout(() => {
+      setIsHideError(true);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    postService.getTodos(USER_ID)
+      .then((todosFromServer: Todo[]) => setTodos(todosFromServer))
+      .catch(() => setError('Unable to get todos'));
+  }, []);
+
+  const activeTodos = useMemo(
+    () => todos.filter(todo => !todo.completed), [todos],
+  );
+
+  const completedTodos = useMemo(
+    () => todos.filter(todo => todo.completed), [todos],
+  );
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = {
+    all: todos,
+    active: activeTodos,
+    completed: completedTodos,
+  };
+
+  const visibleTodos:Todo[]
+    = filteredTodos[filterType as keyof typeof filteredTodos];
```

Comment:
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/96409563/312ccb95-55bc-485d-8c4f-f36119434d48)